records_mover/creds/base_creds.py

```diff
@@ -94,19 +98,23 @@ def default_boto3_session(self) -> Optional['boto3.session.Session']:
             self.__default_boto3_session = self.boto3_session(self._default_aws_creds_name)
         return self.__default_boto3_session
 
+    def _gcp_creds_of_last_resort(self,
+                                  scopes: Iterable[str]) ->\
+            Optional['google.auth.credentials.Credentials']:
+        import google.auth
+        credentials, project = google.auth.default(scopes=scopes)
+        return credentials
```

Comment:
By extracting out this logic to a function, subclasses (like `CredsViaEnv`) can overwrite it and pull credentials from things like environment variables if more specific creds haven't been pointed to with a cred name.